lib/features/modeling/Modeling.js

```diff
@@ -56,7 +56,7 @@ Modeling.prototype.getHandlers = function() {
 };
 
 
-Modeling.prototype.connect = function(source, target, attrs) {
+Modeling.prototype.connect = function(source, target, attrs, hints) {
```

Comment:
Passing hints so that `connectionStart` and `connectionEnd` can be used. Otherwise, it would connect between flow nodes centers. Still, `ConnectionPreview` uses these hints and we could also implement a simple snap-to-center mechanism in order to keep previous behaviour.